### ISSUE TITLE
Unable to read cropped files

### ISSUE THREAD

**Issue by Jean-Du** [AFM-SPM/TopoStats#150]
**Describe the bug**
When attempting to read a cropped spm file, an error is raised (see the output section). This applies to both files manually cropped through gwyddion and files automatically cropped through TopoStats

**To Reproduce**

- Command used to run topostats e.g. `python pygwytracing.py` / `python topostats/topotracing.py`
- Config file (you can attach this or paste it below)
Example of cropped files that gave rise to this error:
https://drive.google.com/file/d/159k-pE8WdlE-_yqqAGkn0sxuYBrp5DHL/view?usp=sharing
(cropped manually in Gwyddion)
https://drive.google.com/file/d/155m9KCemfBvtKCfZQTrq33RfeT7zp9-W/view?usp=sharing
(cropped by TopoStats)

**Expected behavior**

**Output**

If applicable please include the output error, this can be a copy and paste of the output (preferable) or a screenshot.
Output files can be attached to this bug report.

```Traceback (most recent call last):
  File "C:/Users/.../pygwytracing.py", line 794, in <module>
    data = getdata(filename)
  File "C:/Users/.../pygwytracing.py", line 83, in getdata
    data = gwy.gwy_file_load(filename, gwy.RUN_NONINTERACTIVE)
glib.GError: File contains no (importable) data.
```

**(Not required if using Docker) Your computer configuration (please complete the following information):**
 - OS: Windows 10 Pro
 - topostats version: 554b5d6
 - Python version: 2.7
 - Optionally, your installed packages: best way to get this is to paste the results of typing `pip freeze`.

**Additional context**

**Comment by bobturneruk** [comment]
Someone was going to try the new code on `dev`.

**Comment by alicepyne** [comment]
Yes - might be worth uploading a cropped example file to try?

**Comment by bobturneruk** [comment]
Dunno if you can do that in a issue, may need to be a gdrive link (for now , at least).

**Comment by derollins** [comment]
Managed to resolve this issue through changing how cropped images are saved from Gwyddion.

1. Crop the spm file as usual in Gwyddion,
2. Open the Gwyddion data browser (Info, Show Data Browser),
3. Scroll to the bottom of the channel list and select the channel called 'Detail' followed by a number,
4. Extract this to a new file using the button on the bottom left of the Data Browser window,
![Screenshot (42)](https://user-images.githubusercontent.com/69976751/170041292-6eb81c9a-d733-43c4-86e7-c30191f4a53f.png)
5. Moving over to this new file rename the channel, 'Height',
 
![Screenshot (44)](https://user-images.githubusercontent.com/69976751/170041525-3c244de4-b60b-4741-a05c-a59c40a7f251.png)

This cropped file should now be able to be read and analysed by TopoStats.

**Comment by derollins** [comment]
> Wow @derollins - you must've suffered for that knowledge!

Haha thanks. The solution to me came to me all of a sudden but I had been ruminating on it for a couple of weeks.

Guess this is only a solution for the Gwyddion version of Topostats I assume there will be a load of other issues when you try with with Topostats _sans_ Gwyddion.

**Comment by bobturneruk** [comment]
I've seen the same thing and will add some details. I do not expect pySPM will support gwy files. @derollins can you share an example of a cropped file that works with the current version, please?

**Comment by bobturneruk** [comment]
This is my command:

```
python topostats/topotracing.py --config crop/crop.yaml --quiet True
```

config file:

```
base_dir: ./crop/
output_dir: ./crop/output/
save_plots: true
warnings: ignore
cores: 4
quiet: false
file_ext: .spm
channel: Height
amplify_level: 1.0
threshold_method: otsu
grains:
  gaussian_size: 2.0
  gaussian_mode: nearest
  threshold_multiplier: 1.7
  minimum_grain_size: 800
  background: 0.0
```

and output:

```
[Wed, 25 May 2022 09:26:09] [INFO    ] [topostats] Updated config config[quiet] : False > True 
[Wed, 25 May 2022 09:26:09] [INFO    ] [topostats] Configuration file loaded from    : crop/crop.yaml
[Wed, 25 May 2022 09:26:09] [INFO    ] [topostats] Scanning for images in            : crop
[Wed, 25 May 2022 09:26:09] [INFO    ] [topostats] Output directory                  : crop\output   
[Wed, 25 May 2022 09:26:09] [INFO    ] [topostats] Looking for images with extension : .spm
[Wed, 25 May 2022 09:26:09] [INFO    ] [topostats] Images with extension .spm in crop : 1
Processing images from crop, results are under crop\output:   0%|                                                          | 0/1 [00:00<?, ?it/s][Wed, 25 May 2022 09:26:12] [INFO    ] [topostats] Processing : crop\20220413_LinearDNA_339bp_NiCl2_3mM_NaCl_10mM.0_00009.spm
[Wed, 25 May 2022 09:26:12] [INFO    ] [topostats] Extracting filename from : crop\20220413_LinearDNA_339bp_NiCl2_3mM_NaCl_10mM.0_00009.spm
[Wed, 25 May 2022 09:26:12] [INFO    ] [topostats] Filename : 20220413_LinearDNA_339bp_NiCl2_3mM_NaCl_10mM.0_00009
[Wed, 25 May 2022 09:26:12] [INFO    ] [topostats] Loading image from : crop\20220413_LinearDNA_339bp_NiCl2_3mM_NaCl_10mM.0_00009.spm 
```

**Comment by alicepyne** [comment]
As this uses pygwy, is it worth checking with the pygwy forums?

**Comment by ns-rse** [comment]
Closed by #287 which handles non-square images in [TopoStats v2.0.0](https://github.com/AFM-SPM/TopoStats/releases/tag/v2.0.0)
